android/src/main/java/io/customer/reactnative/sdk/extension/TypeConversion.kt

```diff
@@ -20,3 +25,24 @@ internal fun Double?.toCIOLogLevel(fallback: CioLogLevel = CioLogLevel.NONE): Ci
     return if (this == null) fallback
     else CioLogLevel.values().getOrNull(index = toInt() - 1) ?: fallback
 }
+
+/**
+ * Extensions function to build FCM [RemoteMessage] using RN map. This should be independent from
+ * the sender source and should be able to build a valid [RemoteMessage] for our native SDK.
+ *
+ * @param destination receiver of the message. It is mainly required for sending upstream messages,
+ * since we are using RemoteMessage only for broadcasting messages locally, we can use any non-empty
+ * string for it.
+ */
+internal fun ReadableMap.toFCMRemoteMessage(destination: String): RemoteMessage {
+    return with(RemoteMessage.Builder(destination)) {
+        getMap("data")?.toMutableStringMap()?.let { data -> setData(data) }
+        getString("messageId")?.let { id -> setMessageId(id) }
+        getString("messageType")?.let { type -> setMessageType(type) }
+        getString("collapseKey")?.let { key -> setCollapseKey(key) }
+        if (hasKey("ttl")) {
+            ttl = getInt("ttl")
+        }
+        build()
+    }
+}
```

Comment:
We might not need the properties, `messageId`, `messageType`, `collapseKey`, or `ttl` at all, because looking into the implementation seems like `RemoteMessage` object would create themitself based on the `bundle` which is created by the map in the first line, if we cater my first comment, i think this might resolve the use of these keys. 

```
getMap("data")?.toMutableStringMap()?.let { data -> setData(data) }
```

Reply:
Yeah, makes sense. I replaced `setData` with `addData` using iterator for more clarity.